lib/js/src/util/JsonRpcMarshaller.js

```diff
@@ -30,64 +30,68 @@
-var enc = new TextDecoder("utf-8");
-
-undefined
-let step2 = enc.decode(array)
-undefined
-step2
-"{"rpcType":"Request","functionName":"RAI"}"
-let newJson = JSON.parse(step2)
-undefined
-newJson
-{rpcType: "Request", functionName: "RAI"}
-
-
-
-    */
+import { TextEncoder } from './TextEncoder.js';
 
+class JsonRpcMarshaller {
     /**
-     * 
-     * @param {RpcMessage} rpcMessage 
+     * Takes an RPC message and converts it
+     * into a byte array that can be added
+     * as part of the SdlPacket.
+     * @param {RpcStruct} rpcStruct - This param contains the json object to transform
+     * based on the getParameters method. This can be an RpcStruct or something like RpcMessage
+     * which extends RpcStruct.
+     * @returns {null|Array} - Either the byte array or null on error.
      */
-    static marshall(rpcMessage) {
-        //Returns Uint8Array 
+    static marshall (rpcStruct) {
         let jsonBytes = null;
         try {
+            const jsonObject = rpcStruct.getParameters();
+            const stringVersion = JSON.stringify(jsonObject);
+            jsonBytes = this._encode(stringVersion);
+        } catch (error) {
+            console.error('Failed to encode messages to JSON.', error);
+        }
+        return jsonBytes;
+    }
 
-            let jsonObject = rpcMessage.getParameters();
-            //TODO do we need to go through each param and do this?
-            let stringVersion = JSON.stringify(jsonObject);
-            let encoder = new TextEncoder();
-            return encoder.encode(stringVersion);
-
-        } catch (exception) {
-            //TODO print exception
+    /**
+     * Takes an RPC message and converts it
+     * into a byte array that can be added
+     * as part of the SdlPacket.
+     * @param {Array} bytes - Byte array containing the json string to convert.
+     * @returns {null|Object} - Either a json object or null on error in parsing.
+     */
+    static unmarshall (bytes) {
+        let ret = null;
+        try {
+            const jsonString = this._decode(bytes);
+            ret = JSON.parse(jsonString);
+        } catch (error) {
+            console.error('Failed to parse JSON', error);
         }
-        return null;
+        return ret;
     }
 
+    /**
+     * Transform a string into a byte array.
+     * @param {String} str - String to transform.
+     * @returns {Array}
+     * @private
+     */
+    static _encode (str) {
+        const encoder = new TextEncoder();
```

Comment:
Creating a new instance of `TextEncoder` probably doesn't make sense. See other feedback regarding making the TextEncoder's `encode()` method static.

Reply:
updated